### PR TITLE
Longhorn: v1.1.0 release for Rancher v2.5 Chart

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#2061
https://github.com/longhorn/longhorn/issues/1829

### CODE REVIEW

**Comment by shuo-wu** [comment]
How do we test this branch (have an auto-generated from my v2.5 branch) with my private repo?

**Comment by yasker** [comment]
Hi @cbron @aiyengar2 Can you help to review this chart? Thanks.

FYI: @meldafrawi has validated the chart on 2.5 cluster explorer.

**Comment by shuo-wu** [comment]
To test this PR, please use the following URL and branch:
URL `https://github.com/shuo-wu/charts`; Branch `dev-v2.5`

**Review by aiyengar2 — APPROVED** [review]
LGTM, double checked that changes removed from patch were tracked from upstream (left some notes as comments that do not need to be addressed).

Had one minor confirmation, but if the README.md update was intentional this should be good to merge.

File: packages/longhorn/longhorn.patch [hunk 1]
```diff
@@ -50,19 +52,77 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/longhorn/charts-original/Chart.yaml pa
...
+-```
+-helm delete longhorn --purge
+-```
+-
+-With Helm 3 to uninstall Longhorn.
+-```
+-helm uninstall longhorn -n longhorn-system
+-kubectl delete namespace longhorn-system
+-```
++To prevent damage to the Kubernetes cluster, we recommend deleting all Kubernetes workloads using Longhorn volumes (PersistentVolume, PersistentVolumeClaim, StorageClass, Deployment, StatefulSet, DaemonSet, etc).
```
[the remark this answers is not in the document]
Comment (shuo-wu) [reply packages/longhorn/longhorn.patch:106]: We can add this line in the upstream README.md later. But overall the uninstallation guide here is for Rancher v2.5 Chart only.

File: packages/longhorn/longhorn.patch [hunk 2]
```diff
@@ -513,248 +845,76 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/longhorn/charts-original/templates/crd
...
--    engineTag: v1.0.2
--    manager: longhornio/longhorn-manager
--    managerTag: v1.0.2
--    ui: longhornio/longhorn-ui
--    uiTag: v1.0.2
--    instanceManager: longhornio/longhorn-instance-manager
--    instanceManagerTag: v1_20200514
-+    engine:
-+      repository: rancher/longhornio-longhorn-engine
-+      tag: v1.0.2
```
Comment (cbron) [inline packages/longhorn/longhorn.patch:720]: None of these images have changed, is that not required for the release ?

**Review by cbron — COMMENTED** [review]
Seems fine, two questions

File: packages/longhorn/longhorn.patch [hunk 3]
```diff
@@ -50,19 +52,77 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/longhorn/charts-original/Chart.yaml pa
...
+ 
+ ---
+ Please see [link](https://github.com/longhorn/longhorn) for more information.
 diff -x '*.tgz' -x '*.lock' -uNr packages/longhorn/charts-original/crds/crds.yaml packages/longhorn/charts/crds/crds.yaml
 --- packages/longhorn/charts-original/crds/crds.yaml
 +++ packages/longhorn/charts/crds/crds.yaml
-@@ -0,0 +1,172 @@
-+apiVersion: apiextensions.k8s.io/v1beta1
+@@ -0,0 +1,420 @@
++apiVersion: apiextensions.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (shuo-wu) [reply packages/longhorn/longhorn.patch:117]: Yes. The CRD upgrade works fine.

File: packages/longhorn/longhorn.patch (hunk 2 above)
Comment (shuo-wu) [reply packages/longhorn/longhorn.patch:720]: This means no extra modification in `values.yaml` is needed for the new release.

**Review by cbron — APPROVED** [review]

**Pull request merged**